spec/api/landing_applications_spec.rb

```diff
@@ -62,13 +70,19 @@
             example.metadata[:response][:content] = content.deep_merge(example_spec)
           end
 
+          let(:"X-API-KEY") { valid_api_token }
           run_test! do |response|
             data = JSON.parse(response.body)
             expected_data = JSON.parse(expected_json_representation)
 
             expect(data).to eq(expected_data)
           end
         end
+
+        response "401", "invalid credentials" do
+          let(:"X-API-KEY") { "rubbish" }
+          run_test!
+        end
```

Comment:
It's been a while since I looked at Rails obviously - remind me how this test works?

Reply:
A bit of background: these are integration tests which use Rswag's Rspec DSL (domain specific language) . Rswag is the Ruby library we're using to generate our OpenAPI spec (previous known as Swagger) from our specs. We're using the content in our specs as the "example" data in the OpenAPI spec/UI, which allows us to keep both the specs and the Swaggers docs synced with realistic data:

 
![rswag_auto_content](https://github.com/user-attachments/assets/e847bd09-b63b-40bd-9d01-564b4424687f)

Reply:
to answer your question about how that spec works specifically: with

```rb
let(:"X-API-KEY") { "rubbish" }
```
 
we're setting a local variable name `X-API-KEY` and assigning the string "rubbish" to it. This has the effect of setting the HTTP header variable of that name in which the API key is expected to be found. See `swagger.yaml` for how this auth header is defined/configured:

```
components:
  securitySchemes:
    api_key:
      type: apiKey
      name: X-API-KEY
      in: header

```

So the test is asserting that when the auth header doesn't contain a valid key, the response is 401 (unauthorised) with the string "invalid credentials".

See https://github.com/rswag/rswag?tab=readme-ov-file#the-rspec-dsl for details on these RSwag specs.